docs/course_materials/exercises/test_exercises/03_Variables_and_datatypes/test_homework_strings.py

```diff
@@ -18,25 +19,35 @@
 @pytest.mark.strings
 def test_names(helpers):
     helpers.set_student_file('names.py')
-    helpers.expected_pycode = [
+    helpers.expected_stdout = False
+    helpers.tests_list = [
         # Create a string variable with your name, call it my_name
-        [r'''\bmy_name\s*=\s*['"]\w+["']''', 'my_name variable is missing'],
+        [r'''\bmy_name\s*=\s*['"]\w+(\s\w+)*["']''',
+         'check my_name variable definition'],
         # Create a string variable with your family name,
         # call it my_family_name
-        [r'''\bmy_family_name\s*=\s*['"]\w+["']''',
-         'my_family_name variable is missing'],
+        [r'''\bmy_family_name\s*=\s*['"]\w+(\s\w+)*["']''',
+         'check my_family_name variable definition'],
         # Create a string variable called my_full_name which is composed from
         # the 2 variables my_name and my_family_name.
         [r'''\bmy_full_name\s*=.*\bmy_name\b.*\bmy_family_name\b''',
-         'my_full_name variable is not defined'],
+         'expected my_full_name to be defined by the previous variables'],
         # Create a variable with your city name: call it my_city_name
-        [r'''\bmy_city_name\s*=\s*['"]\w+['"]''',
-         'my_city variable is missing'],
+        [r'''\bmy_city_name\s*=\s*['"]\w+(\s\w+)*['"]''',
+         'check my_city_name variable definition'],
         # Create a variable msg with "My name is X and I’m from Y" using
         # the variables you created above
-        [r'''\bmsg\s*=.*[f'My name is {]my_(full_)?name''' +
-         r'''[} and I'm from {my_city_name}]''',
-         'my_message variable is not defined']
+        [r'''\bmy_message\s*=.*my_(full_)?name.*my_city_name.*''',
+         'check my_message variable definition'],
+        [r'''.*f["']My name is \{my_(full_)?name''' +
+         r'''\} and I'm from \{my_city_name\}''',
+         [r'''(.*['"]My name is\s*['"]\s*[+]\s*my_(full_)?name''' +
+          r'''\s*[+]\s*['"]\s*and I'm from\s*['"]\s*[+]\s*''' +
+          r'''my_city_name)|(.*['"]My name is \%s and I'm from \%s''' +
+          r'''['"]\s*\%\s*\(my_(full_)?name,\s*my_city_name\))''',
+          'a better definition for my_message should use f"string"',
+          'expected the use of f"string" for my_message variable, ' +
+          'make sure to include the previous variables']]
```

Comment:
I think a slight alignment is required.

Reply:
Lines 42-50 is one array/answer to be tested.
It tests for the best solution(42-43) and afterwords 2 other possible solutions(44-47).

Reply:
Ack